oauthproxy.go

```diff
@@ -842,6 +844,13 @@ func (p *OAuthProxy) AuthOnly(rw http.ResponseWriter, req *http.Request) {
 
 // SkipAuthProxy proxies allowlisted requests and skips authentication
 func (p *OAuthProxy) SkipAuthProxy(rw http.ResponseWriter, req *http.Request) {
+	if p.addHeadersToSkipped {
+		session, err := p.getAuthenticatedSession(rw, req)
```

Comment:
So using this directly will result in a lot of potential error logging (especially for authenticated but unauthorized code path) that I don't think you want.

I think you really only want this part of that function for your purposes:
```
	var session *sessionsapi.SessionState

	getSession := p.sessionChain.Then(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
		session = middlewareapi.GetRequestScope(req).Session
	}))
	getSession.ServeHTTP(rw, req)
```

Might be worth splitting that out into its own helper method and consuming only that session authentication (without authorization) for this.

Reply:
The one area I'm still leery of - all the underlying session middleware has various logging. The unauthenticated requests might very well trip alot of those and spam the logs. Maybe that's just something users of this optional configuration need to accept?

Reply:
> Might be worth splitting that out into its own helper method and consuming only that session authentication (without authorization) for this.

Good idea, I've done this in 1bcebb2.

> The one area I'm still leery of - all the underlying session middleware has various logging. The unauthenticated requests might very well trip alot of those and spam the logs. Maybe that's just something users of this optional configuration need to accept?

I've tried to tackle this in 1628e22 - I've added a boolean flag to the `RequestScope` saying whether the session is required or optional for this request, and used that to suppress the extraneous logging in the "optional" case.  This was only really one log message that I could find, but now the flag is in the `RequestScope` other places can adapt to it in the same way in future.